kernel/src/syscall/mprotect.rs

```diff
@@ -34,9 +34,11 @@ pub fn sys_mprotect(addr: Vaddr, len: usize, perms: u64, ctx: &Context) -> Resul
     ))?;
     let range = addr..end;
 
-    // On x86, `PROT_WRITE` implies `PROT_READ`.
-    // <https://man7.org/linux/man-pages/man2/mprotect.2.html>
-    #[cfg(target_arch = "x86_64")]
+    // On x86_64 and riscv64, `PROT_WRITE` implies `PROT_READ`.
+    // Reference: <https://man7.org/linux/man-pages/man2/mprotect.2.html>,
+    //            Section 5.11.3 from <https://www.intel.com/content/dam/www/public/us/en/documents/manuals/64-ia-32-architectures-software-developer-vol-3a-part-1-manual.pdf>,
+    //            <https://riscv.github.io/riscv-isa-manual/snapshot/privileged/#translation>.
+    #[cfg(any(target_arch = "x86_64", target_arch = "riscv64"))]
```

Comment:
Use this format:
https://github.com/asterinas/asterinas/blob/fe90e3051b2550b58c71552afa992f2abe59c497/kernel/src/fs/procfs/filesystems.rs#L19-L21

```rust
    // Reference:
    // <https://man7.org/linux/man-pages/man2/mprotect.2.html>,
    // Section 5.11.3 from <https://www.intel.com/content/dam/www/public/us/en/documents/manuals/64-ia-32-architectures-software-developer-vol-3a-part-1-manual.pdf>,
    // <https://riscv.github.io/riscv-isa-manual/snapshot/privileged/#translation>.
```